lib/codegen/src/regalloc/context.rs

```diff
@@ -134,21 +162,48 @@ impl Context {
         );
 
         if isa.flags().enable_verifier() {
-            verify_context(func, cfg, domtree, isa)?;
-            verify_liveness(isa, func, cfg, &self.liveness)?;
-            verify_cssa(func, cfg, domtree, &self.liveness, &self.virtregs)?;
+            let ok = verify_context(func, cfg, domtree, isa, &mut errors).is_ok()
+                && verify_liveness(isa, func, cfg, &self.liveness, &mut errors).is_ok()
+                && verify_cssa(
+                    func,
+                    cfg,
+                    domtree,
+                    &self.liveness,
+                    &self.virtregs,
+                    &mut errors,
+                ).is_ok();
+
+            if !ok {
+                return Err(errors.into());
+            }
         }
 
         // Pass: Coloring.
         self.coloring
             .run(isa, func, domtree, &mut self.liveness, &mut self.tracker);
 
         if isa.flags().enable_verifier() {
-            verify_context(func, cfg, domtree, isa)?;
-            verify_liveness(isa, func, cfg, &self.liveness)?;
-            verify_locations(isa, func, Some(&self.liveness))?;
-            verify_cssa(func, cfg, domtree, &self.liveness, &self.virtregs)?;
+            let ok = verify_context(func, cfg, domtree, isa, &mut errors).is_ok()
+                && verify_liveness(isa, func, cfg, &self.liveness, &mut errors).is_ok()
+                && verify_locations(isa, func, Some(&self.liveness), &mut errors).is_ok()
+                && verify_cssa(
+                    func,
+                    cfg,
+                    domtree,
+                    &self.liveness,
+                    &self.virtregs,
+                    &mut errors,
+                ).is_ok();
+
+            if !ok {
+                return Err(errors.into());
+            }
+        }
+
+        if errors.0.is_empty() {
```

Comment:
Can this condition ever be false? It seems like if we have errors, one of the `if !ok` checks above should have tripped. Can this be an assert instead?

Reply:
It will be `false` if none of the reported errors were fatal. `ok` is `true` iff no fatal error was reported; non-fatal may have been reported, which would make the verification process stop now.